src/index.js

```diff
@@ -259,11 +259,38 @@ const Route = ({ component, url, matches }) => {
 	return h(component, { url, matches });
 };
 
+class AsyncRoute extends Component {
+	constructor() {
+		super();
+		this.state = {
+			componentData: null
+		};
+	}
+	componentDidMount(){
+		const componentData = this.props.component();
+		if (componentData instanceof Promise) {
+			componentData.then(promiseData => {
+				this.setState({
+					componentData: h(promiseData, { url: this.props.url, matches: this.props.matches })
```

Comment:
Suggestion: call `promiseData` `"component"` and store that in state, then invoke `h()` on it in `render()` instead of here.  That will work for updates after initial loading.